ansible_ai_connect/ai/api/formatter.py

```diff
@@ -368,15 +368,33 @@ def get_task_names_from_tasks(tasks):
     return names
 
 
-def restore_original_task_names(output_yaml, prompt):
+def restore_original_task_names(output_yaml, prompt, payload_context=""):
     if output_yaml and is_multi_task_prompt(prompt):
-        prompt_tasks = get_task_names_from_prompt(prompt)
-        matches = re.finditer(r"^- name:\s+(.*)", output_yaml, re.M)
-        for i, match in enumerate(matches):
+        full_yaml = payload_context + output_yaml
+        try:
+            payload_context_data = yaml.safe_load(payload_context)
+            full_data = yaml.safe_load(full_yaml)
+        except Exception as exc:
+            logger.exception(
+                f"Error while loading the result role/playbook YAML:{exc} "
+                f"for restoring the original task names"
+            )
+            return output_yaml
+        prompt_task_names = get_task_names_from_prompt(prompt)
+
+        full_task_list = get_task_list_from_yaml_data_obj(full_data)
+        payload_context_task_list = get_task_list_from_yaml_data_obj(payload_context_data)
+        task_list = []
+
+        for task in full_task_list:
+            if task not in payload_context_task_list:
```

Comment:
It's an edge case but what if there is a task name in the multitask prompt that ends up matching something in the context? Is there another way we could approach this, like maybe assuming the tasks we're updating will always be the last X in the list? Are there scenarios where that doesn't work?

Reply:
Instead of the comparison I can get the task list length from the context and start loping in the whole task list from that index.

The reason I implemented like this is I thought there can be a task added between two for example but yeah we are having the suggestions at the end of the file always, so the item count trick might work. I will update this and ping you.

Reply:
@robinbobbitt applied the mentioned changes. Pls check when available. I want to merge this one in this sprint if no major issues exist.